lib/types.ts

```diff
@@ -18,6 +18,7 @@ export type Note = {
   content: string;
   creationDate: SecondsEpoch;
   deleted: boolean;
+  hasRemoteUpdate?: boolean;
```

Comment:
According to what's written above, and as your IDE should point out if we type `action.note`, this is the wrong spot for `hasRemoteUpdate` since we added it to a `NoteEntity`.

We could:
 - change `Entity` to capture the possibility
 - change `NoteEntity = Entity<Note> & { hasRemoteUpdate?: boolean }`
 - deprecate the need for the property

I'd vote the second or third option.

Reply:
deprecate the need for the property, I think we can do this eventually but right now it is a complex change to make.

Reply:
that's fine. I didn't know if it was or wasn't.